openprescribing/dmd2/forms.py

```diff
@@ -0,0 +1,56 @@
+from django import forms
+
+from crispy_forms.helper import FormHelper
+from crispy_forms.layout import Layout, Field, ButtonHolder, Submit
+from crispy_forms.bootstrap import InlineCheckboxes
+
+
+obj_types_choices = [
+    ('vtm', 'VTMs'),
+    ('vmp', 'VMPs'),
+    ('amp', 'AMPs'),
+    ('vmpp', 'VMPPs'),
+    ('ampp', 'AMPPs'),
+]
+
+include_choices = [
```

Comment:
It's a shame to have to repeat opaque strings like `invalid` throughout the code, but it's tricky to turn these strings into constants elegantly without repetition.  Could this be a case for... metaclasses?

I'm thinking:

```python
class ChoicesType(type):
    def __getattr__(cls, key):
        if key in [x[0] for x in cls.choices]:
            return key
        else:
            raise AttributeError(key)


class DmdLevelChoice(metaclass=ChoicesType):
    choices = [
        ("vtm", "VTMs"),
        ("vmp", "VMPs"),
        ("amp", "AMPs"),
        ("vmpp", "VMPPs"),
        ("ampp", "AMPPs"),
    ]


>>>  print(DmdLevelChoice.vtm)
vtm
>> print(LevelChoice.choices)
[('vtm', 'VTMs'), ('vmp', 'VMPs'), ('amp', 'AMPs'), ('vmpp', 'VMPPs'), ('ampp', 'AMPPs')]
```

Reply:
Nice idea, but I don't think it's worth the extra mental overhead.